qiskit_experiments/library/tomography/fitters/lstsq_utils.py

```diff
@@ -41,96 +43,240 @@ def lstsq_data(
+
+                # Increase counter
+                cond_idxs[outcome_cond] += 1
 
-            # Add vectorized op to stacked basis matrix
-            basis_mat[idx] = np.conj(np.ravel(op, order="F"))
-            probs[idx] = odata[outcome] / shots
-            idx += 1
     return basis_mat, probs
 
 
+def dirichlet_mean_and_var(
+    outcome_data: np.ndarray,
+    shot_data: Optional[Union[np.ndarray, int]] = None,
+    outcome_prior: Union[np.ndarray, int] = 0.5,
+    conditional_measurement_indices: Optional[Sequence[int]] = None,
+) -> Tuple[np.ndarray, np.ndarray]:
+    r"""Compute mean probabilities and variance from outcome data.
+
+    This is computed via a Bayesian update of a Dirichlet distribution
+    with observed outcome data frequences :math:`f_i(s)`, and Dirichlet
+    prior :math:`\alpha_i(s)` for tomography basis index `i` and
+    measurement outcome `s`.
+
+    The mean posterior probabilities are computed as
+
+    .. math:
+        p_i(s) &= \frac{f_i(s) + \alpha_i(s)}{\bar{\alpha}_i + N_i} \\
+        Var[p_i(s)] &= \frac{p_i(s)(1-p_i(s))}{\bar{\alpha}_i + N_i + 1}
+
+    where :math:`N_i = \sum_s f_i(s)` is the total number of shots, and
+    :math:`\bar{\alpha}_i = \sum_s \alpha_i(s)` is the norm of the prior
+    vector for basis index `i`.
+
+    Args:
+        outcome_data: measurement outcome frequency data.
+        shot_data: Optional, basis measurement total shot data. If not
+            provided this will be inferred from the sum of outcome data
+            for each basis index.
+        outcome_prior: measurement outcome Dirichlet distribution prior.
+        conditional_measurement_indices: outcome indices of conditional
+            outcome data.
+
+    Returns:
+        The mean probabilities and variances for Bayesian update
+        with the given outcome data and prior.
+    """
+    size = outcome_data.size
+    dim_circ_cond, num_data, tomo_outcomes = outcome_data.shape
+    if conditional_measurement_indices:
+        dim_cond = 2 ** len(conditional_measurement_indices)
+        f_cond_outcome = _partial_outcome_function(tuple(conditional_measurement_indices))
+    else:
+        dim_cond = 1
+        f_cond_outcome = lambda x: 0
+
+    # Compute hedged probabilities where the "add-beta" rule ensures
+    # there are no zero or 1 values so we don't have any zero variance
+    reduced_size = size // dim_cond // dim_circ_cond
+    mean_probs = np.zeros((dim_circ_cond, dim_cond, reduced_size), dtype=float)
+    var_denom = np.zeros((dim_circ_cond, dim_cond, reduced_size), dtype=int)
+
+    # Bayesian update
+    posterior = outcome_data + outcome_prior
+
+    # Total shots for computing probabilities
+    # If shot data is not provided it is inferred from outcome data
+    # assuming shots is the sum of all observed counts for each basis
+    if shot_data is None:
+        posterior_total = np.sum(posterior, axis=(0, -1))
+    else:
+        # if isinstance(shot_data, np.ndarray):
```

Comment:
Is there any reason to keep unused code?

Reply:
no, this was supposed to be deleted